packages/identity/src/common.ts

```diff
@@ -149,3 +142,29 @@ export const getReqFieldValue = (
     }
   }
 }
+
+export const pubKeyMatch = (cert: string, parsedCsr: CertificationRequest): boolean => {
+  const parsedCertificate = parseCertificate(cert)
+  const pubKey = keyFromCertificate(parsedCertificate)
+  const pubKeyCsr = keyFromCertificate(parsedCsr)
+
+  if (pubKey === pubKeyCsr) {
+    return true
+  }
+  return false
+}
+
+// generalize to certificateByField
+export const certificateByUsername = (username: string, certificates: string[]): string | null => {
+  /**
+   * Check if given username is already in use
+   */
+  for (const cert of certificates) {
+    const parsedCert = parseCertificate(cert)
+    const certUsername = getCertFieldValue(parsedCert, CertFieldsTypes.nickName)
+    if (certUsername?.localeCompare(username, undefined, { sensitivity: 'base' }) === 0) {
```

Comment:
Why are we using localeCompare here?

Reply:
I don't even know what localeCompare is, I just moved the method from backend package to common package without paying attention to its implementation

Reply:
ok, thanks